### PR TITLE
Add a flag to set the SQLite "synchronous" pragma to OFF.

### PR DESCRIPTION
**Pull request by pphaneuf**
And some small cleanups (in a separate commit).

### CODE REVIEW

File: cpp/log/sqlite_db-inl.h [hunk 1]
```diff
@@ -4,12 +4,16 @@
...
 #include "log/sqlite_db.h"
 
+#include <gflags/gflags.h>
 #include <glog/logging.h>
 #include <sqlite3.h>
 
 #include "log/sqlite_statement.h"
 #include "util/util.h"
 
+DEFINE_bool(sqlite_synchronous_off, false,
```
Comment (AlCutter) [inline cpp/log/sqlite_db-inl.h]: Worth a TODO somewhere to capture the conversation we had just now about turning sync off while a fresh node is playing catchup (to let it catch up in less than two weeks), and then turning it on when it's in the game proper?

File: cpp/log/sqlite_db-inl.h [hunk 2]
```diff
@@ -57,6 +61,12 @@ sqlite3* SQLiteOpen(const std::string& dbfile) {
 template <class Logged>
 SQLiteDB<Logged>::SQLiteDB(const std::string& dbfile)
     : db_(SQLiteOpen(dbfile)), tree_size_(0) {
+  if (FLAGS_sqlite_synchronous_off) {
+    sqlite::Statement statement(db_, "PRAGMA synchronous = OFF");
```
Comment (AlCutter) [inline cpp/log/sqlite_db-inl.h]: scary :8ball:

**Comment by AlCutter** [comment]
LGTM

File: cpp/log/sqlite_db-inl.h (hunk 2 above)
Comment (pphaneuf) [inline cpp/log/sqlite_db-inl.h]: And not the default! :smiley:

**Pull request merged**